src/track/taglib/trackmetadata_file.cpp

```diff
@@ -45,36 +45,70 @@ void readAudioProperties(
 
 namespace taglib {
 
+#if QT_VERSION >= QT_VERSION_CHECK(6, 4, 0)
+using namespace Qt::Literals::StringLiterals;
+#else
+constexpr inline QLatin1String operator""_L1(const char* str, size_t size) noexcept {
+    return QLatin1String{str, static_cast<int>(size)};
+}
+#endif
+
 // Deduce the TagLib file type from the file name
-FileType getFileTypeFromFileName(
-        const QString& fileName) {
-    DEBUG_ASSERT(!fileName.isEmpty());
-    const QString fileSuffix(fileName.section(QChar('.'), -1).toLower().trimmed());
-    if (QStringLiteral("mp3") == fileSuffix) {
-        return FileType::MP3;
-    }
-    if ((QStringLiteral("m4a") == fileSuffix) || (QStringLiteral("m4v") == fileSuffix)) {
-        return FileType::MP4;
-    }
-    if (QStringLiteral("flac") == fileSuffix) {
-        return FileType::FLAC;
-    }
-    if (QStringLiteral("ogg") == fileSuffix) {
-        return FileType::OGG;
-    }
-    if (QStringLiteral("opus") == fileSuffix) {
-        return FileType::OPUS;
-    }
-    if (QStringLiteral("wav") == fileSuffix) {
-        return FileType::WAV;
-    }
-    if (QStringLiteral("wv") == fileSuffix) {
-        return FileType::WV;
-    }
-    if (fileSuffix.startsWith(QStringLiteral("aif"))) {
-        return FileType::AIFF;
-    }
-    return FileType::Unknown;
+FileType stringToEnumFileType(
+        const QString& fileType) {
+    DEBUG_ASSERT(!fileType.isEmpty());
+
+    struct TypePair {
+        QLatin1String strType;
+        FileType eType;
+    };
+
+    // This table is aligned with detectByExtension() in fileref.cpp
+    constexpr std::array lookupTable = {
+            TypePair{"mp3"_L1, FileType::MPEG},
```

Comment:
one day we should think about introducing a "minimal perfect hash function"-based approach for LUTs like these.

Reply:
you'll also need to make it static, otherwise the compiler is more likely to choose to create it at runtime. 
```suggestion
    // This table is aligned with detectByExtension() in fileref.cpp
    constexpr static std::array lookupTable = {
            TypePair{"mp3"_L1, FileType::MPEG},
```